app/wama.web/Controllers/UserToolController.cs

```diff
@@ -237,16 +237,19 @@ public IActionResult AddCertification(string memberId)
         {
             SetActiveConsoleTool(Constants.ADMIN_CONSOLE_USERS_PATRONS);
             var Certifications = await _CertificationService.GetCertificationsAsync(memberId);
+            ViewBag.AccountType = await _UserAccountService.GetUserAccountAsync(memberId);
```

Comment:
Although, I feel the name of the property in the **_ViewBag_** could be more precise, since you're storing a **_UserAccount_** in it.

Maybe, change the name to **_ViewBag.UserAccount_**.

Or, perform all the logical checks in the action

```
var account =  await _UserAccountService.GetUserAccountAsync(memberId);
ViewBag.IsPatron = Equals(account, null) ? false : account.AccountType == UserAccountType.Patron;
```